File: SoundMexPro/examples/20 VST Examples/vst_visualize.m

%                    Example for usage of SoundMexPro
% Copyright Hoerzentrum Oldenburg gGmbH Oldenburg 2015, written by Daniel Berg
%
% This example shows the usage of the VST visualization plugin HtVSTVisualize.dll

% call tutorial-initialization script
addpath('..\..\tutorial');
t_00b_init_tutorial

% initialize with default values (two outout channels
if 1 ~= soundmexpro('init', ...                             % command name
        'driver', smpcfg.driver,  ...                       % driver index
        'output', smpcfg.output ...                         % list of output channels to use            
        )
    error(['error calling ''init''' error_loc(dbstack)]);
end

% load visualization plugin to first channel with setting programname
% immediately (i.e. the inifile name to use)
if 1~= soundmexpro('vstload', ...                           % command name
        'filename', '..\..\plugins\htvstvisualize.dll', ...    % VST plugin name
        'input', 0, ...                                     % input(s) to use
        'output', 0, ...                                    % output(s) to use
        'type', 'master', ...                               % plugin type
        'position', 0, ...                                  % plugin position
        'programname', 'visualize.ini' ...                  % program name, which sets inifile name here
        )
    error(['error calling ''vstload''' error_loc(dbstack)]);
end

% load aufio file
if 1 ~= soundmexpro('loadfile', ...                 % command name
        'filename', '../../waves/eurovision.wav', ...  % filename
        'loopcount', 0 ...                          % endless loop 
        )
    error(['error calling ''loadfile''' error_loc(dbstack)]);
end
% go
if 1 ~= soundmexpro('start')
    error(['error calling ''start''' error_loc(dbstack)]);
end

% wait a bit
pause(4);

% hide visualization
if 1 ~= soundmexpro('vstparam',  ...    % command name
        'parameter', 'visible', ...     % parameter name
        'value', 0 ...                  % parameter value
        )
    error(['error calling ''vstparam''' error_loc(dbstack)]);
end

pause(2);
clear soundmexpro;

File: SoundMexPro/examplespy/20 VST Examples/vst_visualize.py

#                    Example for usage of SoundMexPro
# Copyright Hoerzentrum Oldenburg gGmbH Oldenburg 2021, written by Daniel Berg
#
# This example shows the usage of the VST visualization plugin HtVSTVisualize.dll

import sys
import time

# append path to SoundMexPro\bin
sys.path.append('..\\..\\bin')
# import soundmexpro
from soundmexpro import soundmexpro

# NOTE: other than in MATLAB/Octave SoundMexPro raises an error if a command fails, thus
# no need to check a return value after each SoundMexPro command. If you want to change 
# this behaviour please edit ..\bin\soundmexpro.py (look for AssertionError)

# read settings from tutorial
sys.path.append('..\\..\\tutorialpy')
from t_00b_init_tutorial import smp_get_cfg
# get config 
# NOTE: flag 0 will remove inputs, returned dictionary will hold driver and output only, 
# i.e. suitable for 'init' directly
smp_cfg = smp_get_cfg(0)
# init with first two outputs
soundmexpro('init', smp_cfg)    

# load visualization plugin to first channel with setting programname
# immediately (i.e. the inifile name to use)
soundmexpro('vstload',                   # command name
            {
            'filename' : '../../plugins/htvstvisualize.dll', # plugin name
            'input' : 0,                           # inputs to use
            'output' : 0,                          # outputs to use
            'type' : 'master',                          # plugin type
            'position' : 0,                             # plugin position
            'programname' : 'visualize.ini'                   # program name, which sets inifile name here
            }
)


# load audio file
soundmexpro('loadfile',                 # command name
            {
            'filename' : '../../waves/eurovision.wav',  # filename
            'loopcount' : 0                             # endless loop 
            }
)

# go
soundmexpro('start')

# wait a bit
time.sleep(4)

# hide visualization
soundmexpro('vstparam',             # command name
        {
        'parameter' :'visible',     # parameter names
        'value' : 0                 # parameter values
        }
)

time.sleep(2)



